lib/Sdk/OAuth2/ClientCredentials.php

```diff
@@ -30,27 +31,58 @@ function __construct()
      *
      * @throws Throwable If an error occurs during the authentication process.
      */
+
+     private function isValidToken($token): bool
+{
+    return isset($token->access_token) 
+        && isset($token->token_type)
+        && isset($token->expires_in);
+}
+
     public function authenticate(KindeClientSDK $clientSDK, array $additionalParameters = [])
     {
+        $this->storage->setM2MMode(true);
+        error_log('Starting M2M authentication');
+        
         try {
             $client = new Client();
             $formData = [
                 'client_id' => $clientSDK->clientId,
                 'client_secret' => $clientSDK->clientSecret,
-                'grant_type' => GrantType::clientCredentials,
-                'scope' => $clientSDK->scopes
+                'grant_type' => GrantType::clientCredentials
             ];
-            $mergedAdditionalParameters = Utils::addAdditionalParameters($clientSDK->additionalParameters, $additionalParameters);
-            $formData = array_merge($formData, $mergedAdditionalParameters);
-
-            $response =
-                $client->request('POST', $clientSDK->tokenEndpoint, [
-                    'form_params' => $formData
-                ]);
+    
+            if (!empty($clientSDK->scopes)) {
+                $formData['scope'] = $clientSDK->scopes;
+            }
+    
+            $response = $client->request('POST', $clientSDK->tokenEndpoint, [
+                'form_params' => $formData,
+                'headers' => [
+                    'Content-Type' => 'application/x-www-form-urlencoded',
+                    'Accept' => 'application/json'
+                ]
+            ]);
+    
             $token = $response->getBody()->getContents();
-            return json_decode($token);
+            error_log('M2M Token received');
+            $decodedToken = json_decode($token);
+            if ($decodedToken === null && json_last_error() !== JSON_ERROR_NONE) {
+                throw new \RuntimeException('Failed to decode JSON token: ' . json_last_error_msg());
+            }
+            if (!$this->isValidToken($decodedToken)) {
+                 throw new \RuntimeException('Invalid token format received');
+             }
+             
+            $this->storage->setToken($token);
```

Comment:
hey @KeeganBeuthin,
We can't store the client credentials token in storage since it uses cookies which would expose it to end users.

Reply:
hey @davidkinde 
Thank you for replying. I agree the client credentials store most certainly needs to be investigated in that regard.